codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -534,26 +534,50 @@ public static void toHttp2Headers(HttpHeaders inHeaders, Http2Headers out) {
                 if (aName.contentEqualsIgnoreCase(TE)) {
                     toHttp2HeadersFilterTE(entry, out);
                 } else if (aName.contentEqualsIgnoreCase(COOKIE)) {
-                    AsciiString value = AsciiString.of(entry.getValue());
-                    // split up cookies to allow for better compression
-                    // https://tools.ietf.org/html/rfc7540#section-8.1.2.5
+                    CharSequence valueCs = entry.getValue();
                     try {
-                        int index = value.forEachByte(FIND_SEMI_COLON);
-                        if (index != -1) {
-                            int start = 0;
```

Comment:
Feel free to ignore: it was this way when you found it, but if you end up making another commit it might be nice to lift all this out into its own method to isolate the noise. There is already precedent in `toHttp2HeadersFilterTE`.

Reply:
done